.github/workflows/on_pr_master.yml

```diff
@@ -1,14 +1,36 @@
 name: CI/CD
-env:
-  GITHUB_TOKEN: ${{ secrets.PAT_GITHUB_TOKEN }}
 on:
   pull_request:
     branches: [ master ]
     types: [opened, synchronize, reopened, edited, closed]
 
 jobs:
+  cancel-redundant-jobs:
+    runs-on: ubuntu-latest
+    steps:
+      # Use this GitHub Actions at least once in this workflow as soon as possible.
+      - uses: technote-space/auto-cancel-redundant-job@v1
+
+  changes:
+    runs-on: ubuntu-latest
+    # Set job outputs to values from filter step
+    outputs:
+      doc: ${{ steps.filter.outputs.doc }}
+      not-doc: ${{ steps.filter.outputs.not-doc }}
+    steps:
+      # For pull requests it's not necessary to checkout the code
+      - uses: actions/checkout@v2
+      - uses: dorny/paths-filter@v2.2.1
+        id: filter
+        with:
+          filters: |
+            doc:
+              - '**/*.md'
+            not-doc:
+              - '**/!(*.md)'
```

Comment:
I wonder how it will work if PR will contain code and doc changes?

Reply:
Then both will be true :)